diff --git a/Formula/emacs-plus@29.rb b/Formula/emacs-plus@29.rb
index 87b9f0d..c25bb5c 100644
--- a/Formula/emacs-plus@29.rb
+++ b/Formula/emacs-plus@29.rb
@@ -114,6 +114,7 @@ class EmacsPlusAT29 < EmacsBase
     args << "--with-native-compilation" if build.with? "native-comp"
 
     ENV.append "CFLAGS", "-g -Og" if build.with? "debug"
+    ENV.append "CFLAGS", " -DFD_SETSIZE=10000 -DDARWIN_UNLIMITED_SELECT "
 
     args <<
       if build.with? "dbus"
