diff --git a/Formula/gcc.rb b/Formula/gcc.rb
index bcbc53c82..c0732e4b2 100644
--- a/Formula/gcc.rb
+++ b/Formula/gcc.rb
@@ -46,7 +46,7 @@ class Gcc < Formula
     #  - Ada, which requires a pre-existing GCC Ada compiler to bootstrap
     #  - Go, currently not supported on macOS
     #  - BRIG
-    languages = %w[c c++ objc obj-c++ fortran]
+    languages = %w[c c++ objc obj-c++ fortran jit]
 
     osmajor = `uname -r`.split(".").first
     pkgversion = "Homebrew GCC #{pkg_version} #{build.used_options*" "}".strip
@@ -66,6 +66,8 @@ class Gcc < Formula
       --with-system-zlib
       --with-pkgversion=#{pkgversion}
       --with-bugurl=https://github.com/Homebrew/homebrew-core/issues
+      --enable-host-shared
+      --disable-bootstrap          
     ]
 
     # Xcode 10 dropped 32-bit support
@@ -91,7 +93,7 @@ class Gcc < Formula
       # Use -headerpad_max_install_names in the build,
       # otherwise updated load commands won't fit in the Mach-O header.
       # This is needed because `gcc` avoids the superenv shim.
-      system "make", "BOOT_LDFLAGS=-Wl,-headerpad_max_install_names"
+      system "make", "BOOT_LDFLAGS=-Wl,-headerpad_max_install_names -j 8"
       system "make", "install"
 
       bin.install_symlink bin/"gfortran-#{version_suffix}" => "gfortran"
